oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java

```diff
@@ -0,0 +1,406 @@
+	  if (tleResource) {
+
+		logMessage(level, "Title: " + content.getTitle());
+		try {
+		  logMessage(level, "getDataType().getName(): " + content.getDataType().getName());
+		} catch (Exception exc) {
+		  logMessage(1, "Error with getDataType: " + exc.getMessage());
+		}
+
+		ExtendedData extendedData2 = content.getExtendedData();
+		if (extendedData2 != null) {
+		  mapToString(extendedData2.getValues(), level, "Old Extended Data");
+		}
+	  }
+
+	  if (tleResource) {
+		logMessage(0, "Updating:  " + contentDisplay(content));
+		try {
+		  fixContent(courseToc, content, level, placementExternalId);
+		  logMessage(0, "Success");
+		} catch (Exception ex) {
+		  logMessage(0, "Updating ERROR: " + ex.getMessage());
+		  for (StackTraceElement element : ex.getStackTrace()) {
+			logMessage(0, element.toString());
+		  }
+		}
+	  }
+	  recurseContent(contentLoader, courseToc, getChildren(contentLoader, content.getId()), placementExternalId, level + 1);
+	}
+  }
+
+  private void mapToString(Map<String, String> map, int level, String label) {
+	try {
+	  Iterator<String> i = map.keySet().iterator();
+	  while (i.hasNext()) {
+		String attribute = i.next();
+		logMessage(level, attribute + " : " + map.get(attribute));
+	  }
+	} catch (Exception exc) {
+	  logMessage(level, label + " error : " + exc.getMessage());
+	}
+  }
+
+  protected BbList getChildren(ContentDbLoader contentLoader, Id bbContentId) throws PersistenceException {
+	return contentLoader.loadChildren(bbContentId);
+  }
+
+  private void fixContent(CourseToc courseToc, Content ltiContent, int level, String placementExternalId) throws Exception {
+	equellaLookedAt++;
+
+	ExtendedData extendedData = ltiContent.getExtendedData();
+	//Update the URL
+	String newUrl = getEquellaUrl() + extendedData.getValue("url");
+	logMessage(level, "NewUrl: " + newUrl);
+	ltiContent.setUrl(newUrl);
+	//Update the Host
+	String newUrlHost = getDomainName(getEquellaUrl());
+	logMessage(level, "NewUrlHost: " + newUrlHost);
+	ltiContent.getBbAttributes().setString("UrlHost", newUrlHost);
+	//Update the Type
+	ltiContent.setContentHandler("resource/x-bb-blti-link");
+	//Update the content (remove the link)
+	String htmlContent = ltiContent.getBody().getFormattedText();
+	String toRemove = StringUtils.substringBetween(htmlContent, "<div class=\"equella-link", "</div>");
+	htmlContent = StringUtils.remove(htmlContent,toRemove);
+	FormattedText newFormattedText = new FormattedText(htmlContent,FormattedText.Type.HTML);
+	ltiContent.setBody(newFormattedText);
+	//Update the extended Date
+	Map<String, String> values = new HashMap<>();
+	values.put("customParameters", "");
+	values.put("cimPlacementId", placementExternalId);
+	logMessage(level, "PlacementId: " + placementExternalId);
+	values.put("itemOrigin", "CIM");
+	extendedData.setValues(values);
+	mapToString(values,level, "New Extended Data");
+	// persist it
+	contentDbPersister.persist(ltiContent);
+	fixedItems++;
```

Comment:
Consider removing the link from the oEQ database table that the old oEQ LTI B2 uses.